src/backend/src/lib.rs

```diff
@@ -455,13 +458,14 @@ fn add_user_credential(request: AddUserCredentialRequest) -> Result<(), AddUserC
         current_time_ns as u128,
     ) {
         Ok(()) => mutate_state(|s| {
+            let mut user_profile_model =
+                UserProfileModel::new(&mut s.user_profile, &mut s.user_profile_updated);
```

Comment:
I would recommend making this a method of `State`.  Something like `s.user_profile_model() -> UserProfileModel{...}`.  That way `s.user_profile*` can be made private and accessible only via the model.  However that can be in another PR, just to keep the PRs focused.  I didn't mention it before for this reason but it  might be worth mentioning. :-)

Reply:
That's a good idea, I didn't think on that.\

I agree, let's keep it in another PR.

But remember that I'm going on holidays and when I'm back I'll be working mostly on II. I'll try to find time to finish the refactor, but feel free to do so without me.